AEPAnalytics/Sources/Analytics+PublicAPI.swift

```diff
@@ -65,11 +65,18 @@ import Foundation
                 return
             }
 
-            guard let trackingIdentifier = responseEvent.data?[AnalyticsConstants.EventDataKeys.ANALYTICS_ID] as? String else {
-                completion(nil, AEPError.unexpected)
-                return
+            if let responseData = responseEvent.data {
+                if responseData.keys.contains(AnalyticsConstants.EventDataKeys.ANALYTICS_ID) {
```

Comment:
we could group these two into a single if statement 
```
if let responseData = responseEvent.data, responseData.keys.contains(AnalyticsConstants.EventDataKeys.ANALYTICS_ID) 
```
similar for vid

Reply:
Oh yes 👍